03.bowling/bowling.rb

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+X = 'X'
+scores = ARGV[0].split(',')
+frames = []
+process_frames = []
+
+scores.each do |score|
+  if frames.size < 9
+    if score == X
+      frames << [10]
+    else
+      process_frames << score.to_i
+      if process_frames.size == 2
+        frames << process_frames
+        process_frames = []
+      end
+    end
+  else
+    process_frames << if score == X
+                        10
+                      else
+                        score.to_i
+                      end
+  end
+end
+
+frames << process_frames unless process_frames.empty?
+
+total_scores = []
+
+frames.each_with_index do |frame, idx|
```

Comment:
```suggestion
total_score = frames.each_with_index.sum do |frame, idx|
```

↑のように、このループ処理自体を直接sumメソッドと組み合わせることもできます。
公式ドキュメント等で使い方をチェックしてみてください〜。